internal/handler/web/people.go

```diff
@@ -31,3 +34,32 @@ func (h *Handler) GetPeople(ctx *gin.Context) {
 	}
 
 }
+
+func (h *Handler) PostPeopleId(ctx *gin.Context, id string) {
+	var update openapi.PersonUpdate
+	if err := ctx.ShouldBindJSON(&update); err != nil {
+		setInvalid(ctx, err, "Invalid JSON object")
+		return
+	}
+
+	// take the id from the url and get the person
+	person, err := h.users.GetPerson(id)
+	if err != nil {
+		setServerError(ctx, err, "Failed to get person")
+		return
+	}
+
+	if update.TrainingKind != nil {
+		date := *update.TrainingDate
```

Comment:
potential nil-pointer deref here. I'd make the field required in the API definition